sdk/src/dragonboard/local_device.cpp

```diff
@@ -671,3 +683,47 @@ aditof::Status LocalDevice::readLaserTemp(float &temperature) {
 
     return status;
 }
+
+aditof::Status LocalDevice::setCalibrationParams(const std::string &mode,
+                                                 float gain, float offset) {
+    const int16_t pixelMaxValue = (1 << 12) - 1; // 4095
+    CalibrationData calib_data;
+    calib_data.mode = mode;
+    calib_data.gain = gain;
+    calib_data.offset = offset;
+    calib_data.cache = new int16_t[pixelMaxValue];
+    m_implData->calibration_cache[mode] = calib_data;
+
+    for (int16_t current = 0; current < pixelMaxValue; ++current) {
+        calib_data.cache[current] =
+            static_cast<int16_t>(static_cast<float>(current) * gain + offset);
+    }
+
+    return aditof::Status::OK;
+}
+
+aditof::Status LocalDevice::applyCalibrationToFrame(uint16_t *frame,
+                                                    const std::string &mode) {
+
+    auto equal = [](float a, float b) -> bool { return fabs(a - b) < 1e-6; };
+
+    float gain = m_implData->calibration_cache[mode].gain;
+    float offset = m_implData->calibration_cache[mode].offset;
+
+    if (equal(gain, 1.0f) && equal(offset, 0.0f)) {
+        return aditof::Status::OK;
+    }
+
+    unsigned int width = m_implData->frameDetails.width;
+    unsigned int height = m_implData->frameDetails.height;
+
+    int16_t *cache = m_implData->calibration_cache[mode].cache;
+
+    unsigned int size = width * height / 2;
+
+    for (int i = 0; i < size; ++i) {
+        frame[i] = cache[frame[i]];
```

Comment:
This will run faster if the frame pointer is incremented in the for loop and the statement becomes  *frame = cache[*frame];